api/all/src/main/java/io/opentelemetry/api/common/AttributeKeyImpl.java

```diff
@@ -5,12 +5,26 @@
 
 package io.opentelemetry.api.common;
 
-import com.google.auto.value.AutoValue;
 import javax.annotation.Nullable;
 
 @SuppressWarnings("rawtypes")
-@AutoValue
-abstract class AttributeKeyImpl<T> implements AttributeKey<T> {
+final class AttributeKeyImpl<T> implements AttributeKey<T> {
+
+  private final AttributeType type;
+  private final String key;
+  private final int hashCode;
+
+  private AttributeKeyImpl(AttributeType type, String key) {
+    if (type == null) {
+      throw new NullPointerException("Null type");
+    }
+    this.type = type;
+    if (key == null) {
+      throw new NullPointerException("Null key");
+    }
+    this.key = key;
+    this.hashCode = buildHashCode();
```

Comment:
I don't love calling instance method from constructor, but the `EqualsVerifier.withCachedHashCode()` check seems to force that.

Reply:
You can't make it a static method and pass in the fields?

Reply:
oh, well, I think I'd rather skip that equalsverifier check in the test. What is that test doing under the hood anyway?